src/preact/component/renderer.js

```diff
@@ -1,22 +1,28 @@
 import {useLayoutEffect, useState} from '#preact';
 
+/**
+ * @typedef {import('./types').RendererFunction} RendererFunction
+ */
+
+/**
+ * @typedef {import('preact').VNode<*>} VNode
+ */
+
 /**
  * Runs the renderer. It's critical that the `data` argument is stable - it
  * only changes when the actual data changes. The renderer itself is resolved
  * in a layout effect and allows the result to be a promise.
  *
- * @param {?RendererFunctionType|undefined} renderer
+ * @param {RendererFunction} renderer
  * @param {JsonObject} data
- * @return {?RendererFunctionResponseType}
+ * @return {VNode|null}
  */
 export function useRenderer(renderer, data) {
-  const [value, setValue] = useState(
-    /** @type {?RendererFunctionResponseType} */ (null)
-  );
+  const [value, setValue] = useState(/** @type {VNode|null} */ (null));
 
   useLayoutEffect(() => {
     const rendered = (renderer && renderer(data)) || null;
-    if (rendered && typeof rendered['then'] == 'function') {
+    if (rendered && isPromise(rendered)) {
```

Comment:
if `!rendered` then `!isPromise(rendered)`
```suggestion
    if (isPromise(rendered)) {
```

Reply:
Good suggestion. Done